### PR TITLE
Notifications on pipeline failure

### PR DESCRIPTION
**Pull request by elcorbs**
# Description
A lambda function is invoked when an error is caught in the data engineering pipeline. 
The lambda function will send a message to an SNS topic with details of the failure.

I wrapped the whole pipeline in a parallel execution with one branch so that I could catch any error that occurs in there instead of catching it at each individual step. Here is the graph of an execution where an error is caught midway through: 

Screenshot 2022-07-26 at 17 03 57

Emails can be added to the SNS topic in the main workspace in the AWS console to subscribe to these notifications. 

[Trello](https://trello.com/c/lIhtN06J/247-epic4-enable-notifications-if-pipeline-fails-should)

# Developer checklist
- [x] Unit test
- [x] Linked to Trello ticket

### CODE REVIEW

**Review by LauraZM9 — APPROVED** [review]
Looks good!

File: .gitignore [hunk 1]
```diff
@@ -56,3 +56,6 @@ terraform.tfvars
 .python-version
 *.pyc
 .DS_Store
+
+# ------ lambda aritfacts ---- #
```
[the remark this answers is not in the document]
Comment (elcorbs) [reply .gitignore:60]: Apparently artifact is the American spelling, I can change it

Comment (elcorbs) [inline .gitignore:60]:
```suggestion
# ------ lambda artefacts ---- #
```

[the remark this answers is not in the document]
Comment (elcorbs) [reply .gitignore]: Oh, and I hadn't even spelt artifact correctly anyway 🤦‍♀️

**Pull request merged**